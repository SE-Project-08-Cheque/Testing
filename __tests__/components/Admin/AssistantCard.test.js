// import dependencies
import React from "react";

// import react-testing methods
import { render, cleanup } from "@testing-library/react";

// add custom jest matchers from jest-dom
import "@testing-library/jest-dom";

// import react dom
import ReactDOM from "react-dom";

// import renderer for take snapshots
import renderer from "react-test-renderer";

// import the component for testing
import AssistantCard from "../../../components/Admin/AssistantCard";


afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<AssistantCard />, div);
});

it("renders assistant card correctly", () => {
  const { getByTestId } = render(
    <AssistantCard
      imageUrl="./user.png"
      imageAlt="User"
      id="2"
      firstname="chandima"
      lastname="amarasena"
      email="chandima334@gmail.com"
      fetch="fetch"
    />
  );
  expect(getByTestId('display_name')).toHaveTextContent("chandima amarasena");
  expect(getByTestId('email')).toHaveTextContent("chandima334@gmail.com");
});

it("matches snapshot", () => {
  const tree = renderer.create(<AssistantCard
    imageUrl="./user.png"
    imageAlt="User"
    id="2"
    firstname="chandima"
    lastname="amarasena"
    email="chandima334@gmail.com"
    fetch="fetch"
  />);

    expect(tree).toMatchSnapshot();
})

