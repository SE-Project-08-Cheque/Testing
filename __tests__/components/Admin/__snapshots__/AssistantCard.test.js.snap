// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-zn4p7q"
>
  <div
    className="css-19fmqat"
  >
    <div
      className="css-1txosv4"
    >
      <div
        className="css-gmuwbf"
      >
        <img
          alt="User"
          className="chakra-image__placeholder css-1kdrupq"
        />
      </div>
    </div>
    <div
      className="css-1y4dbot"
    >
      <div
        className="css-n16x53"
      >
        <div
          className="css-x1sij0"
        >
          <span
            className="chakra-badge css-637omx"
          >
            New
          </span>
        </div>
        <h4
          className="css-1mwm0bw"
          data-testid="display_name"
        >
          chandima
           
          amarasena
        </h4>
        <div
          className="css-0"
          data-testid="email"
        >
          chandima334@gmail.com
          <span
            className="css-bgvdb6"
          />
        </div>
      </div>
    </div>
    <div
      className="css-dfiqwb"
    >
      <button
        className="chakra-button css-oqfhsh"
        onClick={[Function]}
        type="button"
      >
        Details
        <span
          className="chakra-button__icon css-1oeb2oe"
        >
          <svg
            aria-hidden={true}
            className="chakra-icon css-onkibi"
            focusable={false}
            viewBox="0 0 24 24"
          >
            <path
              d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
              fill="currentColor"
            />
          </svg>
        </span>
      </button>
    </div>
  </div>
</div>
`;
